homeassistant/helpers/device_registry.py

```diff
@@ -732,7 +732,7 @@ def async_config_entry_disabled_by_changed(
 
     if not config_entry.disabled_by:
         for device in devices:
-            if device.disabled_by is not DeviceEntryDisabler.CONFIG_ENTRY:
+            if device.disabled_by != DeviceEntryDisabler.CONFIG_ENTRY:
```

Comment:
It should not be a string?

Reply:
When we load it from JSON it is a string?

Reply:
Yeah, I suspect `device.disabled_by` should not be a string and I saw some depreciation warning if string is being used. Why do we get a string there?

Reply:
Looks like we don't convert it when loading it from storage: https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/device_registry.py#L556

Reply:
We do convert it for Entity Registry https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity_registry.py#L589-L591

Reply:
Ok, now it also makes sense why this didn't blow up more, and why its loading related :sweat_smile: . I'll fix properly.